src/simplewallet/simplewallet.cpp

```diff
@@ -5895,34 +5895,32 @@ bool simple_wallet::transfer_main(Transfer transfer_type, const std::vector<std:
   for (size_t i = 0; i < local_args.size(); )
   {
     dsts_info.emplace_back();
-    cryptonote::address_parse_info & info = dsts_info.back();
+    cryptonote::address_parse_info& info = dsts_info.back();
     cryptonote::tx_destination_entry de;
     bool r = true;
 
-    // check for a URI
-    std::string address_uri, payment_id_uri, tx_description, recipient_name, error;
+    std::string addr, payment_id_uri, tx_description, recipient_name, error;
     std::vector<std::string> unknown_parameters;
     uint64_t amount = 0;
-    bool has_uri = m_wallet->parse_uri(local_args[i], address_uri, payment_id_uri, amount, tx_description, recipient_name, unknown_parameters, error);
-    if (has_uri)
+    bool has_uri = m_wallet->parse_uri(local_args[i], addr, payment_id_uri, amount, tx_description, recipient_name, unknown_parameters, error);
+
+    if (i + 1 < local_args.size())
     {
-      r = cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), address_uri, oa_prompter);
-      if (payment_id_uri.size() == 16)
+      r = cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), local_args[i], oa_prompter);
+      if (!r && m_wallet->is_trusted_daemon())
       {
-        if (!tools::hex_to_type(payment_id_uri, info.payment_id))
-        {
-          fail_msg_writer() << tr("failed to parse short payment ID from URI");
-          return false;
-        }
-        info.has_payment_id = true;
+        rpc::ONS_RESOLVE_ADDRESS::request lookup_req{};
+        lookup_req.address = local_args[i];
+        auto [success, addr_response] = m_wallet->resolve_address(lookup_req);
```

Comment:
As per my earlier comment: this could use `rpc::ONS_RESOLVE` with `type=1`, then decrypt the value here (rather than oxend-side).